core/services/workflows/monitoring.go

```diff
@@ -113,11 +117,17 @@ func initMonitoringResources() (em *engineMetrics, err error) {
 		return nil, fmt.Errorf("failed to register workflow step error counter: %w", err)
 	}
 
+	// Deprecated: use the gauge below
```

Comment:
Can we add this deprecation notice a few other places? Right here isn't the first place that I'd check.

Like on `engineHeartbeatCounter` above in the struct, so if someone tries to use it with code editor autocompletes they will see the inline doc.